nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/versioning/VersioningService.java

```diff
@@ -117,6 +118,21 @@ VersioningOption doPreSave(Document doc, boolean isDirty, VersioningOption optio
             Map<String, Serializable> options);
 
     /**
+     * Applies versioning options before document save.
+     *
+     * @param session the core session
+     * @param doc the document
+     * @param isDirty {@code true} if there is actual data to save
+     * @param option an option chosen by the user or framework
+     * @param checkinComment a checkin comment
+     * @param options map event info
+     * @return the validated option (to use in doPostSave)
+     * @since 9.3
+     */
+    VersioningOption doPreSave(CoreSession session, Document doc, boolean isDirty, VersioningOption option, String checkinComment,
```

Comment:
I'd rather we removed the old version of the interface without a `CoreSession`. These are low-level internal interfaces and we don't need to keep so much compat for a LTS.

Reply:
In this case, can we change `Document` by `DocumentModel` as it was introduced for automatic versioning ? As it we can probably rely on `DocumentModel#followTransition`.

Reply:
You're right that the caller in `AbstractSession` already has a `DocumentModel` available it it may not be necessary to re-instanciate it. Also, as the event handler will potentially modify it, it should be returned to the caller. Or it should be re-read from the `doc` by the caller, like `docModel = readModel(doc)`. Let's think about this a bit more as this is getting complex...